### ISSUE TITLE
Markdown: `md` string literal starting with interpolation does not work correctly

### ISSUE THREAD

**Issue by UtkarshChemE** [JuliaLang/julia#38229]
For the following code:
```julia
begin
	var = 10.0
	md"$(var) value is has newline. But next inline use of $(var) works as expected"
end
```
The output from the same is
```
10.0
value is has newline. But next inline use of 10 works as expected
```

This is a strange behavior where reference of `var` at beginning of string creates a new line, but any other reference doesn't.

**Comment by bahasalien** [comment]
begin
var = 10.0
"$var value is has newline. But next inline use of $var works as expected"
end

done and without "md"....... in REPL

**Comment by Ellipse0934** [comment]
> done and without "md"....... in REPL

No, we are using the `md` macro because we are interested in writing ["julia markdown"(jmd)](https://docs.julialang.org/en/v1/stdlib/Markdown/). Definitely incorrect behavior from the markdown parser. `.jmd` files are used elsewhere too, for example in [Weave.jl](https://github.com/JunoLab/Weave.jl) to create jupyter notebooks,pdf,.etc from jmd files.

**Comment by UtkarshChemE** [comment]
My use case for `md` is Pluto.jl. But either way, I one should expect `md` behavior should be close to non-`md` one.
